backend/main.py

```
@@ -1,538 +1,151 @@
+        # Check limit (e.g., 20 requests per minute)
+        if len(cls._user_requests[client_ip]) >= 20:
+            return False
+        cls._user_requests[client_ip].append(now)
+        return True
 
     @classmethod
     def sanitize_input(cls, text: str) -> str:
-        """Sanitize input text while preserving meaning."""
-        if not isinstance(text, str):
-            return ""
+        if not isinstance(text, str): return ""
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)